core/src/main/java/io/aiven/klaw/service/TopicControllerService.java

```diff
@@ -950,6 +1008,15 @@ public Map<String, Object> getTopicDetailsPerEnv(String envId, String topicName)
     return hashMap;
   }
 
+  public Map<String, String> getAdvancedTopicConfigs() {
+    Map<String, String> topicConfigs = new HashMap<>();
+    for (TopicConfiguration topicConfiguration : TopicConfiguration.values()) {
+      topicConfigs.put(topicConfiguration.name(), topicConfiguration.getValue());
+    }
+
+    return topicConfigs;
+  }
```

Comment:
It generates a new map each time it is called.
It could be moved to `TopicConfiguration` and map could be filled once during class loading as a `Collections.unmodifiedMap(...)`. Then just return same map each time

Reply:
Agree, its part of the same class now